forc-plugins/forc-node/src/chain_config.rs

```diff
@@ -0,0 +1,574 @@
+    pub async fn download_config(&self, conf: &ChainConfig) -> anyhow::Result<()> {
+        let folder_name = match conf {
+            ChainConfig::Local => LOCAL_CONFIG_FOLDER_NAME,
+            ChainConfig::Testnet => TESTNET_CONFIG_FOLDER_NAME,
+            ChainConfig::Ignition => IGNITION_CONFIG_FOLDER_NAME,
+        };
+
+        let api_endpoint = format!(
+            "https://api.github.com/repos/FuelLabs/{}/contents/{}",
+            CHAIN_CONFIG_REPO_NAME, folder_name,
+        );
+
+        let contents = self.fetch_folder_contents(&api_endpoint).await?;
+
+        // Create config directory if it doesn't exist
+        let config_dir = user_forc_directory().join(CONFIG_FOLDER);
+        let target_dir = config_dir.join(folder_name);
+        fs::create_dir_all(&target_dir)?;
+
+        // Download each file
+        for item in contents {
+            if item.content_type == "file" {
+                if let Some(download_url) = item.download_url {
+                    let file_path = target_dir.join(&item.name);
+
+                    let response = self.client.get(&download_url).send().await?;
+
+                    if !response.status().is_success() {
+                        bail!("Failed to download file: {}", item.name);
+                    }
+
+                    let content = response.bytes().await?;
+                    fs::write(file_path, content)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Helper function to fetch folder contents from github.
+    async fn fetch_folder_contents(&self, url: &str) -> anyhow::Result<Vec<GithubContentDetails>> {
+        let response = self
+            .client
+            .get(url)
+            .header("User-Agent", "forc-node")
+            .send()
+            .await?;
+
+        if !response.status().is_success() {
+            bail!("failed to fetch contents from github");
+        }
+
+        Ok(response.json().await?)
+    }
+}
+
+/// Check local state of the configuration file in the vault (if they exists)
+/// and compare them to the remote one in github. If a change is detected asks
+/// user if they want to update, and does the update for them.
+///
+/// Exception is local configuration. For local configurations we are expecting
+/// users to alter their network configs as they see fit. So we only check if
+/// the configuration exists or not, and if it does we do not alter with it.
+/// If the chain config is missing, we are unpacking the one we embedded into
+/// forc-node.
+pub async fn check_and_update_chain_config(conf: ChainConfig) -> anyhow::Result<()> {
+    let fetcher = ConfigFetcher::new();
+    // If chain config is local we will only check if it exists.
+    // If it does not exists we will unpack the one embedded into forc-node.
+    // Otherwise we will continue with what we have in the path without
+    // overriding it.
+    if conf == ChainConfig::Local {
```

Comment:
better to use `match` when checking enum variants

Reply:
I am not interested in other variants.